transcom/mymove
# INT B-20684 858 60 char limit on location names

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20684)

## Summary

When a location name was hecka long, the 858 generation was failing due to constraints within the Go struct. This PR throws a little truncation celebration on the duty location names as well as updates those relevant tests.

### How to test

1. Find a payment request that has the status of `REVIEWED` in DBeaver in the `payment-requests` table
2. We want to change the values of the origin/destination duty location names to something longer than 60 characters to test this
3. Navigate to the orders table by flyin' through the foreign keys as follows
4. `payment_requests` -> `move_id` -> `orders_id`-> `origin_duty_location_id` -> change the `name` column to something long
5. `payment_requests` -> `move_id` -> `orders_id`-> `new_duty_location_id` -> change the `name` column to something long
6. Now go back to the `payment_requests` table on the same row you just navigated through
7. Find the column `payment_request_number`
8. In your CLI, run this command: `go run ./cmd/generate-payment-request-edi/ --payment-request-number <payment_request_number>`
9. You should see a bunch of cryptic looking numbers appear, but the ones we care about are that start with `N1` and you should see the long duty location names truncated and end with `...`

## Screenshots
<img width="1154" alt="Screenshot 2024-07-23 at 1 49 21 PM" src="https://github.com/user-attachments/assets/b105f552-cadc-41d0-af65-2720beb80a8e">
